.eslintrc.json

```diff
@@ -17,30 +26,67 @@
           ],
           // Packages. `react` related packages come first.
           // Also, put `react-icons` in sorting order not with `react`
-          ["^react(?!-.)$", "^\\w", "^@\\w"],
+          [
+            "^react(?!-.)$",
+            "^\\w",
+            "^@\\w"
+          ],
           // The Chakra theme directory if imported to story file or other places
-          ["^@/@chakra-ui"],
+          [
+            "^@/@chakra-ui"
+          ],
           // From the `types` directory.
-          ["^@/lib/types", "^@/lib/interfaces"],
+          [
+            "^@/lib/types",
+            "^@/lib/interfaces"
+          ],
           // From the `components` directory.
-          ["^@/components"],
+          [
+            "^@/components"
+          ],
           // From the `utils` directory.
-          ["^@/lib/utils"],
+          [
+            "^@/lib/utils"
+          ],
           // From the `data` directory.
-          ["^@/data"],
+          [
+            "^@/data"
+          ],
           // From the `constants` directory.
-          ["^@/lib/constants"],
+          [
+            "^@/lib/constants"
+          ],
           // Parent imports. Put `..` last.
-          ["^\\.\\.(?!/?$)", "^\\.\\./?$"],
+          [
+            "^\\.\\.(?!/?$)",
+            "^\\.\\./?$"
+          ],
           // Other relative imports. Put same-folder imports and `.` last.
-          ["^\\./(?=.*/)(?!/?$)", "^\\.(?!/?$)", "^\\./?$"],
+          [
+            "^\\./(?=.*/)(?!/?$)",
+            "^\\.(?!/?$)",
+            "^\\./?$"
+          ],
           // Style imports.
-          ["^.+\\.s?css$"],
+          [
+            "^.+\\.s?css$"
+          ],
           // Side effect imports.
-          ["^\\u0000"]
+          [
+            "^\\u0000"
+          ]
         ]
       }
     ],
-    "simple-import-sort/exports": "error"
+    "simple-import-sort/exports": "error",
+    "no-unused-vars": "off",
+    "@typescript-eslint/no-unused-vars": "off",
```

Comment:
`@typescript-eslint/no-unused-vars` rule overrides `unused-imports/no-unused-vars` where the former will throw an error for an unused param, usually denoted by an underscore.

So for the former, `Array.from({length: 5}).map((_, idx) => {})` will throw an error for the underscore, whereas `unused-imports/no-unused-vars` will know that this is a param ignored intentionally, but still throw a warning or error for declared variables not being used.

Reply:
This is recommended in the [unused-imports eslint docs](https://github.com/sweepline/eslint-plugin-unused-imports?tab=readme-ov-file#usage)

Reply:
Ah got it! thanks

Reply:
Understood your explanation Tyler but I don't see that we are declaring the rule for `unused-imports/no-unused-vars`. I think we need to add that since I don't get any error for unused vars currently.

Reply:
@pettinarip looking at this again, not sure what is up with `unused-imports/no-unused-vars` but can do this instead:
```json
"@typescript-eslint/no-unused-vars": [
  "error", {
    "varsIgnorePattern": "^_"
  }
],
```
To still make sure instance of underscore params are not flagged. This will also not flag any variable that begins with `_`, which will include components.

Should this be an `error` or `warn`?

Reply:
Can we just modify to use `"^_$"` so the `_` must be the first/only/last character for it to match? Then we can make that an error, without it being triggered by the `_` placeholder variables

Reply:
@wackerow @pettinarip I had to review this again and think this through...

So the primary deal here is the use of `unused-imports/no-unused-imports-ts` from `eslint-plugin-unused-imports`, if this did not exist, an unused import would be incorrectly linted under `@typescript-eslint/no-unused-vars`.

With that, `eslint-plugin-unused-imports` recommends this basic rules setup:
```json
rules: {
    "no-unused-vars": "off", // or "@typescript-eslint/no-unused-vars": "off",
    "unused-imports/no-unused-imports": "error",
    "unused-imports/no-unused-vars": [
        "warn",
        {
            "vars": "all",
            "varsIgnorePattern": "^_",
            "args": "after-used",
            "argsIgnorePattern": "^_",
        },
    ]
}
```
And add `"unused-imports/no-unused-imports-ts": "warn"` so a warning shows for the unused import in typescript files.

> Can we just modify to use "^_$" so the _ must be the first/only/last character for it to match? Then we can make that an error, without it being triggered by the _ placeholder variables

You have to turn off `no-unused-vars` and `@typescript-eslint/no-unused-vars` so there are no unexpected overrides.

If you want to be strict to `^_$` and not allow a variable or arg to be named `_variable`, then would need to modify the recommended setup of the `unused-imports/no-unused-vars` rule above:
```json
"unused-imports/no-unused-vars": [
  "error",
    {
    "args": "all",
    "argsIgnorePattern": "^_$",
    "varsIgnorePattern": "^_$"
  }
],
```

Reply:
I don't think we need to not allow an arg like `_variable`, just want to make sure we can still use `_` as an unused placeholder var. Sounds like this is good where it's at.